client/signup/jetpack-connect/index.jsx

```diff
@@ -48,6 +48,17 @@ const MINIMUM_JETPACK_VERSION = '3.9.6';
 const JetpackConnectMain = React.createClass( {
 	displayName: 'JetpackConnectSiteURLStep',
 
+	componentWillMount() {
+		if ( this.props.url ) {
+			let url = this.props.url;
+			if ( url && url.substr( 0, 4 ) !== 'http' ) {
```

Comment:
We're doing that in 2 locations in this file, let's create a utility function/method for that.